pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -300,27 +302,18 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 			var tcpChain, httpChain *listener.FilterChain
 			origDst := svc.GetAddressForProxy(lb.node) + ":" + portString
 			httpClusterName := model.BuildSubsetKey(model.TrafficDirectionInboundVIP, "http", svc.Hostname, port.Port)
+			var filters []*listener.Filter
 			if len(svcAddresses) > 0 && features.EnableAmbientMultiNetwork {
-				setOrigDstForCluster := []*listener.Filter{getOrigDstSfs(origDst, false)}
-				tcpChain = &listener.FilterChain{
-					Filters: append(setOrigDstForCluster, lb.buildInboundNetworkFilters(cc)...),
-					Name:    cc.clusterName,
-				}
-				cc.clusterName = httpClusterName
-				httpChain = &listener.FilterChain{
-					Filters: append(setOrigDstForCluster, lb.buildWaypointInboundHTTPFilters(svc, cc)...),
-					Name:    cc.clusterName,
-				}
-			} else {
-				tcpChain = &listener.FilterChain{
-					Filters: lb.buildInboundNetworkFilters(cc),
-					Name:    cc.clusterName,
-				}
-				cc.clusterName = httpClusterName
-				httpChain = &listener.FilterChain{
-					Filters: lb.buildWaypointInboundHTTPFilters(svc, cc),
-					Name:    cc.clusterName,
-				}
+				filters = []*listener.Filter{getOrigDstSfs(origDst, false)}
+			}
+			tcpChain = &listener.FilterChain{
+				Filters: append(slices.Clone(filters), lb.buildWaypointNetworkFilters(svc, cc)...),
+				Name:    cc.clusterName,
+			}
+			cc.clusterName = httpClusterName
```

Comment:
super-nit: this could be subtly missed when re-arranging things; maybe note `// we're reusing cc here, but now for http`